### PR TITLE
Add AppVeyor file to build Windows installer

### PR DESCRIPTION
**Pull request by X0rg**
I think we are OK about #584.

Things to do **before** accepting this PR (in this order):
- [x] Merge [sjohannes' PyInstaller (`sdk` branch)](https://github.com/sjohannes/pyinstaller/tree/sdk) into [exaile's PyInstaller](https://github.com/exaile/pyinstaller/tree/sdk)
- [x] Create exaile/exaile-binenv-win-appveyor repo from https://github.com/sjohannes/exaile-binenv-win-appveyor
- [x] Setup AppVeyor for Exaile

This `appveyor.yml` file builds and deploys Exaile Windows installer to each new tag.

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+image: Visual Studio 2015
+
+environment:
+  APPVEYOR_SAVE_CACHE_ON_ERROR: "true"
+  GTK_SDK_VERBOSE: 1
+  BUILD_DIR: C:\msys64\home\appveyor
+  BUILD_ROOT: exaile/tools/installer/_build_root
+  SDK_NAME: exaile-binenv-win
+  SDK_VER: 1
+  SDK_URL: https://github.com/exaile/exaile-binenv-win-appveyor/releases/download/$(SDK_NAME)-$(SDK_VER)/$(SDK_NAME)-$(SDK_VER).tar.xz
```
Comment (X0rg) [inline appveyor.yml]: To adapt.

File: appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+artifacts:
+  - path: $(dist_name).exe
+    name: dist
+
+deploy:
+  provider: GitHub
+  release: $(APPVEYOR_REPO_TAG_NAME)
+  description: ""
+  auth_token:
+    secure: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
```
Comment (X0rg) [inline appveyor.yml]: To do.

**Comment by sjohannes** [comment]
> Merge sjohannes' PyInstaller (sdk branch) into exaile's PyInstaller

Done.

> Create exaile/exaile-binenv-win-appveyor repo from https://github.com/sjohannes/exaile-binenv-win-appveyor

I have created [exaile-sdk-win](https://github.com/exaile/exaile-sdk-win) for this, still cleaning up the code a bit before I commit. Then I need to create an AppVeyor project for it.

@virtuald Do we have an AppVeyor team for Exaile?

**Comment by virtuald** [comment]
We don't. I can create one.

**Comment by virtuald** [comment]
Ok, this confuses me, I think I did it wrong. One sec.

File: appveyor.yml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      move "exaile\tools\installer\exaile-LATEST.exe" "%APPVEYOR_BUILD_FOLDER%\%dist_name%.exe"
+
+artifacts:
+  - path: $(dist_name).exe
+    name: dist
+
+deploy:
+  provider: GitHub
+  release: $(APPVEYOR_REPO_TAG_NAME)
+  description: ""
```
Comment (sjohannes) [inline appveyor.yml]: Please add `draft: true` so we can edit the release description first.

(Or if Dustin prefers to create a draft release manually first (before tagging) then we may not want to use automatic deployment at all. But don't worry about that for now.)

**Comment by virtuald** [comment]
Ok, I think I did it right.

File: appveyor.yml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+      bash -lc "time pacman --noconfirm --noprogressbar -Syuu"
+
+before_build:
+  - cmd: |
+      bash -lc "mkdir $BUILD_ROOT"
+      bash -lc "time tar -xf $SDK_NAME-$SDK_VER.tar.xz -C $BUILD_ROOT"
+
+build_script:
+  - cmd: |
+      bash -lc "cd exaile/tools/installer && time ../../../python-gtk3-gst-sdk/win_installer/build_win32_installer.sh"
```
Comment (sjohannes) [inline appveyor.yml]: I wonder if setting `MAKEFLAGS=-j` can speed this up a bit.

File: appveyor.yml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+skip_non_tags: true
+
+version: "{branch}-{build}"
+
+image: Visual Studio 2015
+
+environment:
+  APPVEYOR_SAVE_CACHE_ON_ERROR: "true"
+  GTK_SDK_VERBOSE: 1
+  BUILD_DIR: C:\msys64\home\appveyor
```
[the remark this answers is not in the document]
Comment (X0rg) [reply appveyor.yml]: 093279c

File: appveyor.yml [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+clone_folder: $(BUILD_DIR)\$(APPVEYOR_PROJECT_NAME)
+
+clone_depth: 1
+
+cache:
+  - C:\msys64\var\cache\pacman\pkg
+  - '%BUILD_DIR%\%SDK_NAME%-%SDK_VER%.tar.xz'
+
+init:
+  - set PATH=C:\msys64\usr\bin;%PATH%
```
[the remark this answers is not in the document]
Comment (X0rg) [reply appveyor.yml]: 0b13640

File: appveyor.yml (hunk 3 above)
Comment (X0rg) [reply appveyor.yml]: 644c3e6

File: appveyor.yml (hunk 1 above)
Comment (X0rg) [reply appveyor.yml]: 8c4b53f

File: appveyor.yml (hunk 4 above)
Comment (X0rg) [reply appveyor.yml]: I dunno, you can try. It seems the VM has 2 CPUs, why not.

**Comment by X0rg** [comment]
https://github.com/exaile/exaile-sdk-win/releases/download/exaile-sdk-win-6/exaile-sdk-win-6.tar.xz :ok_hand:

File: appveyor.yml (hunk 4 above)
Comment (X0rg) [reply appveyor.yml]: [Before](https://ci.appveyor.com/project/X0rg/exaile/builds/22960607#L1347):
```
real	4m7.466s
user	0m20.471s
sys	0m39.088s
```
[After](https://ci.appveyor.com/project/X0rg/exaile/builds/22964889#L1658):
```
real	4m19.881s
user	0m21.022s
sys	0m46.107s
```

Comment (X0rg) [reply appveyor.yml]: [Before](https://ci.appveyor.com/project/X0rg/exaile/builds/22960607#L1347):
```
real	4m7.466s
user	0m20.471s
sys	0m39.088s
```
[After](https://ci.appveyor.com/project/X0rg/exaile/builds/22965096#L1398):
```
real	4m27.253s
user	0m22.522s
sys	0m45.907s
```

With `set MAKEFLAGS=-j0`, [it was complaining with `the '-j' option requires a positive integer argument`](https://ci.appveyor.com/project/X0rg/exaile/builds/22964889#L608).
With `MAKEFLAGS=-j$(nproc)`, there is no `make[2]` in log. It uses only one thread, while [`nproc` is `2` on this VM](https://ci.appveyor.com/project/X0rg/exaile/builds/22965010#L6).

Comment (X0rg) [reply appveyor.yml]: Well, added `MAKEFLAGS=-j2` in 6785443.

**Comment by virtuald** [comment]
So.. we think this will just work then? I suppose we should do another rc release anyways. I added a project for this on appveyor, but it only builds tags (which this appveyor config specifies also) so I guess we can't test it without a tag?

**Comment by sjohannes** [comment]
Yeah, I think this currently needs a tag (and only on master?). We can do one-off tests with throwaway tags, but if we want to do actual CI this will need some rethinking.

We can probably make it so that it runs builds on every commit, but only creates an artifact and release on tags.

File: appveyor.yml (hunk 2 above)
Comment (sjohannes) [reply appveyor.yml]: You can use [my auth token](https://github.com/exaile/exaile-sdk-win/blob/master/appveyor.yml#L70) (please keep the comment). Not sure if we're meant to reuse the same token across projects, but I think it should be safe.

**Comment by X0rg** [comment]
@virtuald: You need to set `auth_token` to deploy artifact. https://www.appveyor.com/docs/deployment/github/#provider-settings

@sjohannes: Done in 43bd721. It produces a draft with artifact that can be edited and then published.

**Comment by sjohannes** [comment]
Nice, it integrates with GitHub as a check.

Is there a good way to use a different name for [the artifact](https://ci.appveyor.com/project/ExaileDevelopmentTeam/exaile/builds/22973741/artifacts) if it's a non-tag build?

**Comment by X0rg** [comment]
@sjohannes: Fixed in d6e7f03.
I use `APPVEYOR_BUILD_NUMBER` but [some others are available](https://www.appveyor.com/docs/environment-variables/).

File: appveyor.yml (hunk 2 above)
Comment (X0rg) [reply appveyor.yml]: d7d590c4

**Comment by X0rg** [comment]
Oops, build fixed in b7c9bde.
Enjoy https://ci.appveyor.com/project/ExaileDevelopmentTeam/exaile/builds/22977876/artifacts

**Comment by X0rg** [comment]
@sjohannes: You are welcome. You have been a great help to me you too.

**Comment by virtuald** [comment]
I also really appreciate the work you guys have put into this. Even though I was the one who originally introduced Windows releases, it really had become the most painful part of the release process. This makes it much easier.

**Pull request merged**